sqlmesh/integrations/dlt.py

```diff
@@ -0,0 +1,169 @@
+    """
+
+    import dlt
+    from dlt.common.schema.utils import has_table_seen_data, is_complete_column
+    from dlt.pipeline.exceptions import CannotRestorePipelineException
+
+    try:
+        pipeline = dlt.attach(pipeline_name=pipeline_name)
+    except CannotRestorePipelineException:
+        raise click.ClickException("Could not attach to pipeline {pipeline_name}")
+
+    schema = pipeline.default_schema
+    dataset = pipeline.dataset_name
+
+    client = pipeline._sql_job_client(pipeline.default_schema)
+    config = client.config
+    credentials = config.credentials
+    db_type = pipeline.destination.to_name(pipeline.destination)
+    configs = {
+        key: value
+        for key in dir(credentials)
+        if not key.startswith("_")
+        and not callable(value := getattr(credentials, key))
+        and value is not None
+    }
+
+    dlt_tables = {
+        name: table
+        for name, table in schema.tables.items()
+        if (has_table_seen_data(table) and not name.startswith(schema._dlt_tables_prefix))
+        or name == schema.loads_table_name
+    }
+
+    sqlmesh_models = set()
+    sqlmesh_schema_name = f"{dataset}_sqlmesh"
+    for table_name, table in dlt_tables.items():
+        dlt_columns = {
+            c["name"]: exp.DataType.build(str(c["data_type"]), dialect=dialect)
+            for c in filter(is_complete_column, table["columns"].values())
+            if c.get("name") and c.get("data_type")
+        }
+        primary_key = [
+            str(c["name"])
+            for c in filter(is_complete_column, table["columns"].values())
+            if c.get("primary_key") and c.get("name")
+        ]
+        model_def_columns = format_columns(dlt_columns, dialect)
+        select_columns = format_columns_for_select(dlt_columns)
+        grain = format_grain(primary_key)
+        incremental_model_name = f"{sqlmesh_schema_name}.incremental_{table_name}"
+        full_model_name = f"{sqlmesh_schema_name}.full_{table_name}"
+
+        incremental_model_sql = generate_incremental_model(
+            incremental_model_name,
+            model_def_columns,
+            select_columns,
+            grain,
+            dataset + "." + table_name,
+        )
+        full_model_sql = generate_full_model(
+            full_model_name, model_def_columns, select_columns, grain, incremental_model_name
+        )
+
+        sqlmesh_models.update(
+            {(incremental_model_name, incremental_model_sql), (full_model_name, full_model_sql)}
+        )
+
+    return sqlmesh_models, format_config(configs, db_type)
+
+
+def generate_incremental_model(
+    model_name: str, model_def_columns: str, select_columns: str, grain: str, from_table: str
+) -> str:
+    """Generate the SQL definition for an incremental model."""
+
+    key = "load_id" if from_table.endswith("_dlt_loads") else "_dlt_id"
+
+    return f"""MODEL (
+  name {model_name},
+  kind INCREMENTAL_BY_UNIQUE_KEY (
```

Comment:
Is it always unique by key?

Reply:
I thought this was the simpler way since each dlt table has a unique load id column and use this as its key. And if someone wanted to implement some other kind of table they could adjust these later. But I can change it if you feel another type should serve better as a default generated table

Reply:
shouldn't you filter by dates here? otherwise you're always doing a full scan on from_table

Reply:
Altered the model kinds to `INCREMENTAL_BY_TIME_RANGE` by using the load time (converting the load ids which are unix timestamps) as time_column